src/backend/distributed/commands/role.c

```diff
@@ -0,0 +1,128 @@
+#include "postgres.h"
+
+#include "nodes/parsenodes.h"
+#include "utils/syscache.h"
+#include "access/table.h"
+#include "catalog/pg_authid.h"
+#include "utils/rel.h"
+#include "utils/builtins.h"
+#include "access/heapam.h"
+#include "distributed/worker_transaction.h"
+#include "distributed/commands/utility_hook.h"
+#include "distributed/master_protocol.h"
+
+List * ProcessAlterRoleStmt(AlterRoleStmt *stmt, const char *queryString);
+static const char * CreateSetPasswordQueryForWorkerNodes(const char *rolename);
+List * CreateSetPasswordQueriesOfUsersWithPassword(void);
+
+
+bool EnablePasswordPropagation = true;
+
+/*
+ * ProcessAlterRoleStmt actually creates the plan we need to execute for alter
+ * role statement.
+ */
+List *
+ProcessAlterRoleStmt(AlterRoleStmt *stmt, const char *queryString)
+{
+	ListCell *optionCell = NULL;
+	List *commands = NIL;
+	const char *setPasswordQuery = NULL;
+
+	if (!IsCoordinator())
```

Comment:
I tried to make this function a little more generic. It is for all `ALTER ROLE` statements, instead of just `ALTER ROLE ... PASSWORD` statements. So if we ever want to also propagate, let's say, `ALTER ROLE ... CREATEROLE` statements we will just have to add one more `if` in the `foreach`. So I put the check in the the `if`.